microcloud/cmd/microcloud/ask.go

```diff
@@ -740,25 +740,27 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 					}
 
 					ipConfig[gateway] = fmt.Sprintf("%s-%s", rangeStart, rangeEnd)
-
-					gatewayAddr, _, err := net.ParseCIDR(gateway)
-					if err != nil {
-						return err
-					}
-
-					dnsAddresses, err := c.asker.AskString(fmt.Sprintf("Specify the DNS addresses (comma-separated IPv4 / IPv6 addresses) for the distributed network (default: %s): ", gatewayAddr.String()), gatewayAddr.String(), validate.Optional(validate.IsListOf(validate.IsNetworkAddress)))
-					if err != nil {
-						return err
-					}
-
-					if dnsAddresses != "" {
-						dnsConfig[gateway] = dnsAddresses
-					}
 				} else {
 					ipConfig[gateway] = ""
 				}
 			}
 		}
+
+		gateways := []string{}
+		for gateway := range ipConfig {
+			gatewayAddr, _, err := net.ParseCIDR(gateway)
```

Comment:
Why do you run `net.ParseCIDR` here? Is it for the purpose of validating that the gateway is valid? I think the validation is already done in the step before when invoking `AskString`.

Below you again get the string from the `net.IP` and append it to the list of gateways. So could you just take the gateway from the map and append it to the list?

Reply:
Nevermind, you want to extract the IP only without the subnetmask.